### DISCUSSION TITLE
Rerun testbench without restarting

### DISCUSSION THREAD

**Discussion by patrick-5546** [cocotb/cocotb#4155] [Q&A] [answered]
I am using the modelsim/questa GUI to run my testbench. Is there a way to rerun it besides closing the window and running it again? I tried `restart -f` then `run -all` but that doesn't work, perhaps because it doesn't restart the cocotb portion

**Comment by ktbarrett** [comment, marked as answer]
This is not a currently supported feature.

**Comment by elgorwi** [comment]
@patrick-5546 you can relaunch the sim from the modelsim tcl prompt:
`do runsim.do`

**Reply by patrick-5546** [reply]
What is in runsim.do? I am using cocotb-test so I might not have that file

**Reply by elgorwi** [reply]
@patrick-5546 `runsim.do` is a generated file:
https://github.com/cocotb/cocotb/blob/5d0f4f7677858c611a3de212d6c978fdd939b925/src/cocotb_tools/makefiles/simulators/Makefile.questa#L119

The file will be generated when you use the makefile to launch the sim.

Have not yet been able to get this to work with cocotb-test

**Reply by patrick-5546** [reply]
I ended up overriding the `Questa` class to generate runsim.do from a jinja template. Thanks for the idea!
